File: Php/app/Cashgame/Action/BuyinModel.php
namespace app\Cashgame\Action{

	use core\Globalization;
	use core\HomegamePageModel;
	use entities\Cashgame;
	use Domain\Classes\User;
	use entities\CashgameResult;
	use entities\Homegame;
	use app\Urls\CashgameBuyinUrlModel;
	use entities\Player;

    class BuyinModel extends HomegamePageModel {

		/** @var CashgameResult */
		private $result;
		public $buyinUrl;
		public $stackFieldEnabled;
		public $buyinAmount;

        public function __construct(User $user,
									Homegame $homegame,
									Player $player,
									array $years = null,
									Cashgame $runningGame,
									$postedAmount = null){
			parent::__construct($user, $homegame, $runningGame);
            $this->result = $runningGame->getResult($player);
			$this->buyinUrl = new CashgameBuyinUrlModel($homegame, $player);
			$this->stackFieldEnabled = $runningGame->isInGame($player);
			$this->buyinAmount = $postedAmount != null ? $postedAmount : $homegame->getDefaultBuyin();
        }

	}

}

File: Web/Models/CashgameModels/Buyin/BuyinModel.cs
using Core.Classes;
using Web.Models.PageBaseModels;
using Web.Models.UrlModels;

namespace Web.Models.CashgameModels.Buyin{

    public class BuyinModel : HomegamePageModel {

        public UrlModel BuyinUrl { get; set; }
        public bool StackFieldEnabled { get; set; }
		public int BuyinAmount { get; set; }
        public int StackAmount { get; set; }

        public BuyinModel(User user, Homegame homegame, Player player, Cashgame runningGame, int? postedAmount = null)
            :base(user, homegame, runningGame)
        {
			BuyinUrl = new CashgameBuyinUrlModel(homegame, player);
			StackFieldEnabled = runningGame.IsInGame(player);
			BuyinAmount = postedAmount.HasValue ? postedAmount.Value : homegame.DefaultBuyin;
            StackAmount = 0;
        }

        public override string BrowserTitle
        {
            get
            {
                return "Buy In";
            }
        }

	}

}